### ISSUE TITLE
Error: cannot find value `out` in this scope when deriving `Archive` for `enum` in macro

### ISSUE THREAD

**Issue by DarrenTsung** [rkyv/rkyv#252]
Hi, I'm upgrading to the latest rkyv release from a pretty old version (~0.4) and found this strange bug when I try to generate enums from a macro that have `#[derive(Archive)]` attached.

Here's a minimal repo case that errors with rkyv 0.7.32:
```
use rkyv::*;

macro_rules! generate_enums {
    ($($enum_name:ident$(,)*)*) => {
        $(
            #[derive(Archive)]
            pub enum $enum_name {
                HasInput,
            }
        )*
    };
}

// compiler errors here
generate_enums!(Foo, Baz);

// doesn't error when hand written
#[derive(Archive)]
pub enum Bar {
    HasInput,
}
```

I also pushed this repro case to a repository here: https://github.com/DarrenTsung/rkyv-clippy-repro.

**Comment by djkoloski** [comment]
Thanks for the bug report, I'm taking a look now.

**Comment by djkoloski** [comment]
This should be fixed by 6bd0eb360b1f636c81151fc26ef7694422db9745. I was using `quote_spanned` in many places to try to improve error messaging, but this interacts poorly with macro hygiene since it uses the span location to determine if macro hygiene has been violated. This is what caused the derive to emit really weird errors about `out` not being defined despite it being an argument to a derive-generated impl. Can you make sure that fixes your use case completely?

**Comment by DarrenTsung** [comment]
Yep, that fixed it for me thanks!
